activesupport/lib/active_support/callbacks.rb

```diff
@@ -747,12 +803,25 @@ def define_callbacks(*names)
           options = names.extract_options!
 
           names.each do |name|
-            class_attribute "_#{name}_callbacks", instance_writer: false
+            name = name.to_sym
+
             set_callbacks name, CallbackChain.new(name, options)
 
             module_eval <<-RUBY, __FILE__, __LINE__ + 1
```

Comment:
I don't think we need to. They should not be public API, so if we don't need inside the framework we can remove without deprecation.